src/common/graphHelpers.js

```diff
@@ -50,154 +102,113 @@ const getLabel = ({ data, previousData, formattedDate, label, previousLabel }) =
   return `${updatedLabel}\n ${previousCount > -1 ? '+' : ''}${previousCount} ${previousLabel}`;
 };
 
+// ToDo: when the API returns filler date values "fillFormatChartData" should be updated
 /**
- * Fills missing dates with zeroed y-values, updates labels
- * walk the date range and fill missing values, all the while updating previous labels
+ * Fill missing dates, and format graph data
  *
- * @param startDate {string}
- * @param endDate {string}
- * @param values {object} pre-filled key-value object
- * @param label {string} i18n specific label
- * @param previousLabel {string} i18n specific previousLabel
+ * @param {Array} data
+ * @param {Date} endDate
+ * @param {string} granularity, see enum of rhelApiTypes.RHSM_API_QUERY_GRANULARITY_TYPES
+ * @param {Date} startDate
  * @returns {Array}
  */
-const fillMissingValues = ({ startDate, endDate, values, label, previousLabel }) => {
-  const endDateStartDateDiff = moment(endDate).diff(startDate, 'days');
+const fillFormatChartData = ({ data, endDate, granularity, startDate }) => {
+  const granularityType = getGranularityDateType(granularity);
+  const endDateStartDateDiff = moment(endDate).diff(startDate, granularityType);
   const chartData = [];
 
+  let previousData = null;
+  let previousYear = null;
+
   for (let i = 0; i <= endDateStartDateDiff; i++) {
-    const formattedDate = moment
-      .utc(startDate)
-      .add(i, 'days')
-      .format(chartDateFormat);
-    const updatedLabel = getLabel({
-      data: values[formattedDate] ? values[formattedDate].y : 0,
-      previousData: i > 0 ? chartData[i - 1].y : null,
-      formattedDate,
-      label,
-      previousLabel
-    });
+    if (GRANULARITY_TYPES.QUARTERLY === granularity && i % 4 !== 0) {
+      continue;
+    }
+
+    const momentDate = moment.utc(startDate).add(i, granularityType);
+    const stringDate = momentDate.toISOString();
+    const year = parseInt(momentDate.year(), 10);
 
-    if (values[formattedDate]) {
-      chartData.push({
-        x: values[formattedDate].x,
-        y: values[formattedDate].y,
-        label: updatedLabel
-      });
+    const isNewYear = i !== 0 && year !== previousYear;
+    let formattedDate;
+
+    if (granularityType === 'months') {
```

Comment:
constant for `months`?

Reply:
same as above, these are from the `getGranularityDateType` aimed specifically at moment